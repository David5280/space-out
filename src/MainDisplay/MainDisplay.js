import React from 'react';
import InfoContainer from '../InfoContainer/InfoContainer';

const MainDisplay = ({ data }) => {
  const dataToDisplay = data.map(datum => {
    return (
      <InfoContainer 
      description={datum.description}
      imagePreview={datum.image}/>
    )
  })
  return (
    <section className='main-section-display'>
      {data && dataToDisplay}
    </section>
  )
}

export default MainDisplay;