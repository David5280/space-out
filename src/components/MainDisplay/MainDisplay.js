import React from 'react';
import { Route } from 'react-router-dom';
import PlanetCutout from '../PlanetCutout/PlanetCutout';
import PlanetInfo from '../PlanetInfo/PlanetInfo';

const MainDisplay = ({ planets }) => {
  const dataToDisplay = planets.map(planet => {
    return (
      <PlanetCutout 
        title={planet.title}
        cutout={planet.cutout}
        key={planet.key}
      />
    )
  });
  const routes = planets.map(planet => {
    return (
      <Route 
        exact path={`/${planet.title}`} 
        render={() => <PlanetInfo planet={planet}
        />
      } key={planet.key}
      />
    )
  });
  return (
    <section className='main-section-display'>
    {routes}
    {planets && <Route exact path='/' render={() => dataToDisplay} />}
    </section>
  )
};

export default MainDisplay;