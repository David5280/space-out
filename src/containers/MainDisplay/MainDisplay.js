import React from 'react';
import PlanetCutout from '../../components/PlanetCutout/PlanetCutout';
import PlanetInfo from '../../components/PlanetInfo/PlanetInfo';
import UserStats from '../../components/UserStats/UserStats';
import StatsForm from '../StatsForm/StatsForm';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const MainDisplay = ({ planets, user }) => {
  const dataToDisplay = planets.map(planet => {
    return (
      <PlanetCutout 
        title={planet.title}
        cutout={planet.cutout}
        key={planet.key}
      />
    )
  });
  const routes = planets.map(planet => {
    return (
      <div key={planet.title}>
        <Route 
          exact path={`/${planet.title}`} 
          render={() => <PlanetInfo planet={planet}
        />
        } 
        />
        <Route 
          exact path={`/stats/${planet.title}`} 
          render={() => <UserStats planet={planet} user={user}
        />
        } 
        />
      </div>
    )
  });
  const statsForm = () => {
    return <StatsForm />
  };
  return (
    <main className='main-section-display'>
    {routes}
    {planets && <Route exact path='/' render={() => dataToDisplay} />}
    <Route exact path='/enter-stats' component={statsForm} />
    </main>
  )
};

const mapStateToProps = (state) => ({
  user: state.user
});

MainDisplay.propTypes = {
  planets: PropTypes.array
}

export default connect(mapStateToProps)(MainDisplay);