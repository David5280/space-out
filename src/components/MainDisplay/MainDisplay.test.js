import React from 'react';
import { shallow } from 'enzyme';
import MainDisplay from './MainDisplay';

describe('Main Display', () => {
  let wrapper;
  let mockPlanets;
  beforeEach(() => {
    mockPlanets = [
      {title: 'earth', cutout: '/fakecutout',  key: '2'}
    ];
    wrapper = shallow(<MainDisplay planets={mockPlanets} />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})