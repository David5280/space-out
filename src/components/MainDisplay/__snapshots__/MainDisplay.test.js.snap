// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Display should match the snapshot 1`] = `ShallowWrapper {}`;
